oauth2client/service_account.py

```diff
@@ -461,3 +462,56 @@ def create_delegated(self, sub):
         result._private_key_pkcs12 = self._private_key_pkcs12
         result._private_key_password = self._private_key_password
         return result
+
+
+class SelfSignedJWTCredentials(object):
+    """Self signed JWT credentials.
+
+    Makes an assertion to server using a self signed JWT from service account
+    credentials.  These credentials do NOT use OAuth 2.0 and instead authenticate
+    directly.
+    """
+    MAX_TOKEN_LIFETIME_SECS = 3600
+    """Default lifetime of the token (one hour, in seconds)."""
+
+    def __init__(self, creds, audience, time_limit):
+        self._creds = creds
+        self._audience = audience
```

Comment:
I don't see why you couldn't use the same JWT Credentials instance to encode jwts for multiple audiences. I feel like this should be an argument on `encode`.